### ISSUE TITLE
[🐛 Bug]: selenium-server fails to launch intermittently

### ISSUE THREAD

**Issue by romovs** [SeleniumHQ/selenium#11725]
### What happened?

We are experiencing a rather strange behavior with selenium-server intermittently failing to start.
Basically we have an application which launches selenium using the following cmd:
```
java -jar -Djcommander.debug=true -Dwebdriver.ie.driver=wIEDriverServer.exe -Dwebdriver.gecko.driver="geckodriver.exe" -Dwebdriver.edge.driver="msedgedriver.exe" -Dwebdriver.chrome.driver="chromedriver.exe" selenium-server-4.8.1.jar standalone --port 4444 --session-timeout 1500 --override-max-sessions true --max-sessions 100
```

And 1 in 10 or so times, selenium fails to start with the following error. This seems to be highly system dependent and on some systems happens 1 in 10 times, on other 1 in 1000:
```
Was passed main parameter '--port' but no main parameter was defined in your arg class
```

After enabling `jcommander.debug` there seems to be a difference JCommander debug logs. `org.openqa.selenium.grid.server.BaseServerFlags@50cbc42f ` is missing in case of a failed launch.

**Successful launch:**
```
[JCommander] Parsing "--port 4444 --session-timeout 1500 --override-max-sessions true --max-sessions 100"
  with:org.openqa.selenium.grid.server.HelpFlags@4d76f3f8 
org.openqa.selenium.grid.config.ConfigFlags@2d8e6db6 
org.openqa.selenium.grid.node.config.NodeFlags@5ddfd24d 
org.openqa.selenium.grid.server.BaseServerFlags@50cbc42f 
org.openqa.selenium.grid.node.docker.DockerFlags@15d0aca4 
org.openqa.selenium.grid.router.httpd.RouterFlags@75412c2f 
org.openqa.selenium.grid.node.relay.RelayFlags@45538b14 
org.openqa.selenium.grid.distributor.config.DistributorFlags@182d6294 
org.openqa.selenium.grid.log.LoggingFlags@3b438666 
org.openqa.selenium.grid.sessionqueue.config.NewSessionQueueFlags@282ba1e
[JCommander] Adding description for -h
[JCommander] Adding description for -help
[JCommander] Adding description for --help
[JCommander] Adding description for /?
[JCommander] Adding description for --version
[JCommander] Adding description for --config
[JCommander] Adding description for --dump-config
[JCommander] Adding description for --config-help
[JCommander] Adding description for --max-sessions
[JCommander] Adding description for --override-max-sessions
[JCommander] Adding description for --session-timeout
[JCommander] Adding description for --detect-drivers
[JCommander] Adding description for -I
[JCommander] Adding description for --driver-implementation
[JCommander] Adding description for --driver-factory
[JCommander] Adding description for --grid-url
[JCommander] Adding description for --hub
[JCommander] Adding description for --driver-configuration
[JCommander] Adding description for --register-cycle
[JCommander] Adding description for --register-period
[JCommander] Adding description for --heartbeat-period
[JCommander] Adding description for --vnc-env-var
[JCommander] Adding description for --no-vnc-port
[JCommander] Adding description for --drain-after-session-count
[JCommander] Adding description for --enable-cdp
[JCommander] Adding description for --enable-bidi
[JCommander] Adding description for --node-implementation
[JCommander] Adding description for --downloads-path
[JCommander] Adding description for --host
[JCommander] Adding description for --bind-host
[JCommander] Adding description for -p
[JCommander] Adding description for --port
[JCommander] Adding description for --max-threads
[JCommander] Adding description for --allow-cors
[JCommander] Adding description for --https-private-key
[JCommander] Adding description for --https-certificate
[JCommander] Adding description for --registration-secret
[JCommander] Adding description for --self-signed-https
[JCommander] Adding description for --docker-url
[JCommander] Adding description for --docker-host
[JCommander] Adding description for --docker-port
[JCommander] Adding description for --docker
[JCommander] Adding description for -D
[JCommander] Adding description for --docker-devices
[JCommander] Adding description for --docker-video-image
[JCommander] Adding description for --docker-assets-path
[JCommander] Adding description for --relax-checks
[JCommander] Adding description for --username
[JCommander] Adding description for --password
[JCommander] Adding description for --sub-path
[JCommander] Adding description for --service-configuration
[JCommander] Adding description for --service-url
[JCommander] Adding description for --service-host
[JCommander] Adding description for --service-port
[JCommander] Adding description for --service-status-endpoint
[JCommander] Adding description for -d
[JCommander] Adding description for --distributor
[JCommander] Adding description for --distributor-port
[JCommander] Adding description for --distributor-host
[JCommander] Adding description for --distributor-implementation
[JCommander] Adding description for --slot-matcher
[JCommander] Adding description for --slot-selector
[JCommander] Adding description for --healthcheck-interval
[JCommander] Adding description for --reject-unsupported-caps
[JCommander] Adding description for --newsession-threadpool-size
[JCommander] Adding description for --configure-logging
[JCommander] Adding description for --structured-logs
[JCommander] Adding description for --plain-logs
[JCommander] Adding description for --tracing
[JCommander] Adding description for --http-logs
[JCommander] Adding description for --log
[JCommander] Adding description for --log-encoding
[JCommander] Adding description for --log-level
[JCommander] Adding description for --log-timestamp-format
[JCommander] Adding description for --sq
[JCommander] Adding description for --sessionqueue
[JCommander] Adding description for --sessionqueue-port
[JCommander] Adding description for --sessionqueue-host
[JCommander] Adding description for --session-request-timeout
[JCommander] Adding description for --session-request-timeout-period
[JCommander] Adding description for --session-retry-interval
[JCommander] Adding description for --sessionqueue-batch-size
[JCommander] Parsing arg: --port
[ParameterDescription] Adding value:4444 to parameter:port
[JCommander] Parsing arg: --session-timeout
[ParameterDescription] Adding value:1500 to parameter:sessionTimeout
[JCommander] Parsing arg: --override-max-sessions
[ParameterDescription] Adding value:true to parameter:overrideMaxSessions
[JCommander] Parsing arg: --max-sessions
[ParameterDescription] Adding value:100 to parameter:maxSessions
01:47:26.972 INFO [LoggingOptions.configureLogEncoding] - Using the system default encoding
```

**Failed launch:**

```
[JCommander] Parsing "--port 4444 --session-timeout 1500 --override-max-sessions true --max-sessions 100"
  with:org.openqa.selenium.grid.server.HelpFlags@4d76f3f8 
org.openqa.selenium.grid.config.ConfigFlags@2d8e6db6 
org.openqa.selenium.grid.node.config.NodeFlags@5ddfd24d 
org.openqa.selenium.grid.node.docker.DockerFlags@15d0aca4 
org.openqa.selenium.grid.router.httpd.RouterFlags@75412c2f 
org.openqa.selenium.grid.node.relay.RelayFlags@45538b14 
org.openqa.selenium.grid.distributor.config.DistributorFlags@182d6294 
org.openqa.selenium.grid.sessionqueue.config.NewSessionQueueFlags@282ba1e 
org.openqa.selenium.grid.log.LoggingFlags@3b438666
[JCommander] Adding description for -h
[JCommander] Adding description for -help
[JCommander] Adding description for --help
[JCommander] Adding description for /?
[JCommander] Adding description for --version
[JCommander] Adding description for --config
[JCommander] Adding description for --dump-config
[JCommander] Adding description for --config-help
[JCommander] Adding description for --max-sessions
[JCommander] Adding description for --override-max-sessions
[JCommander] Adding description for --session-timeout
[JCommander] Adding description for --detect-drivers
[JCommander] Adding description for -I
[JCommander] Adding description for --driver-implementation
[JCommander] Adding description for --driver-factory
[JCommander] Adding description for --grid-url
[JCommander] Adding description for --hub
[JCommander] Adding description for --driver-configuration
[JCommander] Adding description for --register-cycle
[JCommander] Adding description for --register-period
[JCommander] Adding description for --heartbeat-period
[JCommander] Adding description for --vnc-env-var
[JCommander] Adding description for --no-vnc-port
[JCommander] Adding description for --drain-after-session-count
[JCommander] Adding description for --enable-cdp
[JCommander] Adding description for --enable-bidi
[JCommander] Adding description for --node-implementation
[JCommander] Adding description for --downloads-path
[JCommander] Adding description for --docker-url
[JCommander] Adding description for --docker-host
[JCommander] Adding description for --docker-port
[JCommander] Adding description for --docker
[JCommander] Adding description for -D
[JCommander] Adding description for --docker-devices
[JCommander] Adding description for --docker-video-image
[JCommander] Adding description for --docker-assets-path
[JCommander] Adding description for --relax-checks
[JCommander] Adding description for --username
[JCommander] Adding description for --password
[JCommander] Adding description for --sub-path
[JCommander] Adding description for --service-configuration
[JCommander] Adding description for --service-url
[JCommander] Adding description for --service-host
[JCommander] Adding description for --service-port
[JCommander] Adding description for --service-status-endpoint
[JCommander] Adding description for -d
[JCommander] Adding description for --distributor
[JCommander] Adding description for --distributor-port
[JCommander] Adding description for --distributor-host
[JCommander] Adding description for --distributor-implementation
[JCommander] Adding description for --slot-matcher
[JCommander] Adding description for --slot-selector
[JCommander] Adding description for --healthcheck-interval
[JCommander] Adding description for --reject-unsupported-caps
[JCommander] Adding description for --newsession-threadpool-size
[JCommander] Adding description for --sq
[JCommander] Adding description for --sessionqueue
[JCommander] Adding description for --sessionqueue-port
[JCommander] Adding description for --sessionqueue-host
[JCommander] Adding description for --session-request-timeout
[JCommander] Adding description for --session-request-timeout-period
[JCommander] Adding description for --session-retry-interval
[JCommander] Adding description for --sessionqueue-batch-size
[JCommander] Adding description for --configure-logging
[JCommander] Adding description for --structured-logs
[JCommander] Adding description for --plain-logs
[JCommander] Adding description for --tracing
[JCommander] Adding description for --http-logs
[JCommander] Adding description for --log
[JCommander] Adding description for --log-encoding
[JCommander] Adding description for --log-level
[JCommander] Adding description for --log-timestamp-format
[JCommander] Parsing arg: --port
Was passed main parameter '--port' but no main parameter was defined in your arg class
Usage: standalone [options]
  Options:
```

### How can we reproduce the issue?
Execute following the batch:
```shell
for /l %%x in (1, 1, 500) do (
  start "" /B java -jar -Djcommander.debug=true -Dwebdriver.chrome.driver="chromedriver.exe" selenium-server-4.8.0.jar standalone --port 4444 --session-timeout 1500 --override-max-sessions true --max-sessions 100
  timeout 5 > nul	
  taskkill /im java.exe /f
)
```

### Relevant log output

```shell
Logs are included in the message body.
```

### Operating System

Windows 10

### Selenium version

4.8.1

### What are the browser(s) and version(s) where you see this issue?

n/a

### What are the browser driver(s) and version(s) where you see this issue?

n/a

### Are you using Selenium Grid?

4.8.1

**Comment by diemol** [comment]
Please provide at least the code you use to launch the server. If it happens intermittently, we should be able to loop the command and get the message, right?

**Comment by krmahadevan** [comment]
@romovs 

Adding to what @diemol asked for...

You mentioned 
> Basically we have an application which launches selenium using the following cmd:
> And 1 in 10 or so times, selenium fails to start with the following error. This seems to be highly system dependent and on some systems happens 1 in 10 times, on other 1 in 1000:
> 

> Was passed main parameter '--port' but no main parameter was defined in your arg class

My hunch is that the application that you are using to launch the server seems to be having some issues owing to which the port number never gets generated and passed.

* So you would need to show us, what is this application that you are using.
* Also have you tried adding some debugging statements which will print the raw values that the application generates (am guessing that you perhaps are also computing the port number to spawn the selenium server) before it passes them as parameters to the selenium server spawning command.

If the port number is always constant, then the command you shared should have had the actual value.

**Comment by romovs** [comment]
@diemol @krmahadevan The application is rather large and not trivial to setup. However I'll try to prepare a simplified example.

We verified that all parameters are passed correctly from the application. JCommander (I am referring to JCommander used in selenium-server of course) debug printouts in selenium-server confirm this (first line in the attached logs). 

Also сelenium-server was recompiled with some additional logging. The arguments seemed fine everywhere we looked. The only suspicious moment is the missing mention of `org.openqa.selenium.grid.server.BaseServerFlags@50cbc42f ` in JCommander logs. Unfortunately we weren't able to debug this any further due to lack of necessary Java expertise.

**Comment by romovs** [comment]
I am able to reproduce this more or less consistently by running the following batch:
```
for /l %%x in (1, 1, 500) do (
  start "" /B java -jar -Djcommander.debug=true -Dwebdriver.chrome.driver="chromedriver.exe" selenium-server-4.8.0.jar standalone --port 4444 --session-timeout 1500 --override-max-sessions true --max-sessions 100
  timeout 5 > nul	
  taskkill /im java.exe /f
)
```

Attached is the log. See line 930+.
[log.txt](https://github.com/SeleniumHQ/selenium/files/10869719/log.txt)

java -version
```
openjdk version "19.0.2" 2023-01-17
OpenJDK Runtime Environment (build 19.0.2+7-44)
OpenJDK 64-Bit Server VM (build 19.0.2+7-44, mixed mode, sharing)
```

**Comment by romovs** [comment]
I have tested with previous Java versions and the issue is not reproducible with any versions prior to 19. Happens only with Java 19.

**Comment by diemol** [comment]
Any specific reason to use Java 19? We'd probably have a look right away if this was failing in the LTS version. It'd be great if you can find the time to give us a hand researching this as well.

**Comment by joerg1985** [comment]
#12382 was a duplicate of this and i have fixed it with #12384, i must have missed this issue.
